packages/airnode-node/src/evm/fulfillments/index.ts

```diff
@@ -9,141 +7,147 @@ import {
   Request,
   EVMProviderState,
   ProviderState,
-  RequestStatus,
   RequestType,
   TransactionOptions,
-  TransactionReceipt,
+  LogsErrorData,
+  GroupedRequests,
+  ApiCall,
+  Withdrawal,
 } from '../../types';
-import { AirnodeRrpFactory } from '../contracts';
+import { AirnodeRrpFactory, AirnodeRrp } from '../contracts';
 import * as verification from '../verification';
 
-export interface Receipt {
-  readonly id: string;
-  readonly data?: string;
-  readonly error?: Error;
-  readonly type: RequestType;
-}
-
-interface RequestReceipt {
-  readonly id: string;
-  readonly type: RequestType;
-  readonly data?: Transaction;
-  readonly err?: Error | null;
-}
-
-interface OrderedRequest {
+interface OrderedRequest<T> {
   readonly nonce: number;
-  readonly makeRequest: () => Promise<RequestReceipt>;
+  type: RequestType;
+  readonly makeRequest: () => Promise<Request<T>>;
 }
 
-/**
- * Sponsor requests are performed from the sponsor wallet. Each transaction has a fixed nonce and blochchain rejects out
- * of order transactions.
- *
- * This function makes all the sponsor requests sequentially (ordered by transaction nonce increasingly).
- *
- * There is a concept of batched requests, but that doesn't work with transactions. See:
- * https://github.com/ethers-io/ethers.js/issues/892#issuecomment-828897859
- */
-const submitSponsorRequestsSequentially = async (state: ProviderState<EVMProviderState>, sponsorAddress: string) => {
+function baseLogOptions(state: ProviderState<EVMProviderState>) {
```

Comment:
This function could maybe be extracted to somewhere more generic (it's not specific to fulfilments). I think it could be re-used in some places

Reply:
> This function could maybe be extracted to somewhere more generic (it's not specific to fulfilments). I think it could be re-used in some places

You're right but I would leave that for a separate refactor PR.